### ISSUE TITLE
aws_transfer_server returns not-working public endpoint when deployed with VPC_ENDPOINT type

### ISSUE THREAD

**Issue by mkielar** [hashicorp/terraform-provider-aws#9014]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform Version
```
Terraform v0.12.1
+ provider.aws v2.13.0
```

### Affected Resource(s)

* aws_transfer_server

### Terraform Configuration Files

```hcl
resource "aws_transfer_server" "this" {
  identity_provider_type = "SERVICE_MANAGED"
  endpoint_type          = "VPC_ENDPOINT"
  logging_role           = aws_iam_role.server_logging_role.arn

  endpoint_details {
    vpc_endpoint_id = var.transfer_server_vpc_endpoint_id
  }

resource "aws_route53_record" "this" {
  name    = var.sftp_domain_name
  zone_id = var.sftp_zone_id
  type    = "CNAME"
  ttl     = 60

  records = [
    aws_transfer_server.this.endpoint # will point to not-working public endpoint
  ]
}
```

### Expected Behavior

The route53 entry is created with the working endpoint of the Transfer Server (which, in case of the `VPC_ENDPOINT` deployment, would be the first domain on the `dns_entry` output of `aws_vpc_endpoint` resource, which ID is passed in `endpoint_details`.

### Actual Behavior

The public endpoint of the Transfer Server is returned, which does not resolve to anything meaningful when trying to connect.

Trying to `sftp` to such server ends up with:
```
ssh: Could not resolve hostname sftp.example.com: Name or service not known
Connection closed
```

### Steps to Reproduce

1. `terraform apply`

### Important Factoids

Possible workaround:
```hcl
resource "aws_route53_record" "this" {
  name    = var.sftp_domain_name
  zone_id = var.sftp_zone_id
  type    = "CNAME"
  ttl     = 60

  records = [
    var.sftp_endpoint_type == "VPC_ENDPOINT"
    ? aws_vpc_endpoint.transfer_server.dns_entry[0].dns_name  # Use VPC Endpoint if deployed that way
    : aws_transfer_server.this.endpoint  # Use transfer server public endpoint otherwise
  ]
}
```

### References

* #0000

**Comment by nywilken** [comment]
Hi @mkielar let me know if I misunderstanding the report, but are you trying to create a transfer server that is publicly accessible? If so, have you tried setting the `endpoint_type` argument set to `PUBLIC`, as in `endpoint_type="PUBLIC"`. This will result in a publicly accessible endpoint address, which you should be able to CNAME to in the route53 hosted zone. 

If there is any other relevant information to help us figure out if this is a bug please add it to the report. Otherwise, if `endpoint_type = PUBLIC` is the setting you were looking for please feel free to close this issue. 

Cheers.

**Comment by mkielar** [comment]
Hi @nywilken,

Ok, that might have been confusing, so for clarification:
- When I refer to `VPC_ENDPOINT`, I mean the type of Transfer Server deployment (as opposed to `PUBLIC`)
- When I refer to "VPC Endpoint", I mean the Interface Endpoint to Transfer Server that I had to deploy separately in my VPC (I think it's called PrivateLink?).

So, to recap:
What I really did is I deployed the `aws_transfer_server` with `PUBLIC` type, and then changed it to `VPC_ENDPOINT`, after realizing `PUBLIC` is not what I wanted.

So it's the `VPC_ENDPOINT` what I really need. However, I still want to have pretty domain name for it in Route53. The goal is, to have this Transfer Server available in a VPN, and I want to provide the other side of the VPN connection with a pretty domain name to my sftp. 

So, I want to deploy an internal Route53 record, that would point to a Transfer Server deployed in a VPC.
The problem is, if I do that, and then use the `endpoint` attribute of the `aws_transfer_server` resource, it still gives me the url to public endpoint, as if I was deploying with `PUBLIC` type, not `VPC_ENDPOINT` type.

The correct behaviour (IMO) would be, for the `endpoint` attribute to return the working url of the Transfer Server in both (`PUBLIC` and `VPC_ENDPOINT` cases).

So, when the Transfer Server is deployed with `VPC_ENDPOINT` type, the `endpoint` attribute should return the URL of the VPC Endpoint to Transfer Server, instead of the not-working with public internet url.
